b_cfn_opensearch_index/source/main/action.py

```diff
@@ -0,0 +1,216 @@
+            boto3_session=boto3.Session(),
+            opensearch_endpoint=self.opensearch_domain_endpoint
+        )
+
+    def create(self) -> ActionResult:
+        """
+        Creates an empty index applying index settings and mappings from invocation event Resource Properties.
+
+        :return: Create action result.
+        """
+
+        logger.info(f'Initiating resource creation with these properties: {json.dumps(self.__properties)}.')
+
+        # Create index name by concatinating index prefix and url safe random string.
+        index_name = f'{self.index_prefix}-{token_urlsafe(8)}'.lower()
+
+        self.__create(
+            index_name=index_name,
+            body={
+                'settings': self.index_settings,
+                'mappings': self.index_mapping_settings
+            }
+        )
+
+        return ActionResult(data={'IndexName': index_name}, resource_id=f'{self.opensearch_domain_endpoint}||{index_name}')
+
+    def update(self):
+        """
+        Update settings and mapping of existing index.
+
+        Only dynamic index settings can be updated. On try to update static index settings an exception will be raised.
+
+        :return: Update action result.
+        """
+
+        # List of static index settings. These settings cannot be updated on live index.
+        # https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/create-index/
+        static_index_settings = [
+            'index.number_of_shards',
+            'index.number_of_routing_shards',
+            'index.shard.check_on_startup',
+            'index.codec',
+            'index.routing_partition_size'
+            'index.soft_deletes.retention_lease.period',
+            'index.load_fixed_bitset_filters_eagerly',
+            'index.hidden'
+        ]
+
+        logger.info(f'Initiating resource update with these properties: {json.dumps(self.__properties)}.')
+
+        # Extracing index name from PhysicalResourceId.
+        index_name = self.__resource_id.split('||')[1]
+        # Extracting old index prefix from custom resource old properties.
+        old_index_prefix = self.__old_properties['IndexPrefix']
+
+        # In case new index prefix diffears old one create new index with new name. Index name cannot be updated.
+        # CloudFormation will delete old index together with all documents which were indexed using this index.
+        if self.index_prefix != old_index_prefix:
+            new_index_name = f'{self.index_prefix}-{token_urlsafe(8)}'.lower()
+            self.__create(
+                index_name=new_index_name,
+                body={
+                    'settings': self.index_settings,
+                    'mappings': self.index_mapping_settings
+                }
+            )
+            # In case custom resource property Reindex explicitly set True.
+            # Try to reindex existing documents using new index.
+            if self.__properties['Reindex']:
```

Comment:
IF - ELSE statement too big and does not look elegant. You probably want to have separate functions.